### ISSUE TITLE
Editing cells with specific predefined values

### ISSUE THREAD

**Issue by ricnov** [man-group/dtale#659]
Hi, is it possible to edit cells with predefined specific values (not data type)? For example in a column with string data type it allows only "yes", "no" or "maybe". Thanks

**Comment by aschonfeld** [comment]
@ricnov still working on supporting your full functionality, but just released v2.10.0 with an update where if your column is a category.  It will allow you to choose from the different categories when editing.  So for your example you could create a categorical column with values (yes, no, maybe).

I'll keep working on this feature so that you're not limited to categorical columns.  Will keep you posted

**Comment by aschonfeld** [comment]
@ricnov I'm about to merge an update to support this functionality.  Here is how it would work:
```
import pandas as pd
import dtale

# Option 1 -> using 'column_edit_options' parameter
dtale.show(pd.DataFrame(dict(foo=['yes'])), column_edit_options=dict(foo=['yes', 'no', 'maybe']))

# Option 2 -> updating settings on instance
d = dtale.show(pd.DataFrame(dict(foo=['yes'])))
d.update_settings(column_edit_options=dict(foo=['yes',' no', 'maybe']))

# Option 3 -> updating settings using instance ID
from dtale.views import _update_settings

d = dtale.show(pd.DataFrame(dict(foo=['yes'])))
_update_settings(d._data_id, dict(column_edit_options=dict(foo=['yes', 'no', 'maybe'])))
```

Let me know if you have any issues with this or tweaks you'd like made.  If I don't hear anything I'll move forward with this change and let you know when a new version is out there

**Comment by aschonfeld** [comment]
@ricnov just released v2.11.0 with this feature.  Please let me know if you have any issues.  I've also added another option for  setting options for column editing. Following this [documentation](https://github.com/man-group/dtale/blob/master/docs/CONFIGURATION.md) you can add the following to the `[show]` section for it to be applied to every dataframe you load into D-Tale without specifying parameters in your code:
```
column_edit_options = {"foo": ["yes", "no", "maybe"]}
```

Lastly, if you haven't already please toss your ⭐ on the repo.  Thanks 🙏
